File: INTERSECTION-ARRAYS/ia.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        s1 = set(nums1)
        s2 = set(nums2)
        arr = []
        for i in s1:
            if i in s2:
                arr.append(i)
        return arr

File: INTERSECTION-ARRAYS/ia.java
import java.util.*;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        for (int n : nums1) {
            set1.add(n);
        }

        Set<Integer> set2 = new HashSet<>();
        for (int n : nums2) {
            set2.add(n);
        }

        List<Integer> result = new ArrayList<>();
        for (int n : set1) {
            if (set2.contains(n)) {
                result.add(n);
            }
        }

        int[] arr = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            arr[i] = result.get(i);
        }

        return arr;
    }
}
